plugins/outputs/opensearch/opensearch_test.go

```diff
@@ -0,0 +1,733 @@
+package opensearch
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const servicePort = "9200"
+
+func launchTestContainer(t *testing.T) *testutil.Container {
+	container := testutil.Container{
+		Image:        "opensearchproject/opensearch:1.1.0",
```

Comment:
Thank you for the integration tests. There are 2.x tags as well. I would like to see us run tests on both if possible. What do you think?

Reply:
Hi @powersj , have added tests for V2 as well.